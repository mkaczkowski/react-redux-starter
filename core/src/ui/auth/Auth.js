// @flow
import * as React from 'react';
import type { ActionType } from '@core/constants/flowTypes';
import CONFIG from '@core/config';

export type AuthProps = {
  children?: any,
  isLoading?: boolean,
  actions?: {
    authAction: ActionType,
  },
};

class Auth extends React.Component<AuthProps> {
  componentDidMount() {
    const { actions } = this.props;

    //TODO remove data
    actions &&
      actions.authAction({
        email: CONFIG.api.userLogin,
        password: CONFIG.api.userPassword,
        remember_me: '0',
      });
  }

  render() {
    return this.props.isLoading ? 'Loading....' : this.props.children;
  }
}

export default Auth;
