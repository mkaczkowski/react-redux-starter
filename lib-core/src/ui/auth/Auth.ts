import * as React from 'react';
// import type { ActionType } from 'core/src/constants/flowTypes';
import CONFIG from 'lib-core/src/config';

export type AuthProps = {
  children?: any;
  isLoading?: boolean;
  actions?: {
    authAction: ActionType;
  };
};

class Auth extends React.Component<AuthProps> {
  componentDidMount() {
    const { actions } = this.props;
    //TODO remove data
    if (actions) {
      actions.authAction({
        email: CONFIG.api.userLogin,
        password: CONFIG.api.userPassword,
        remember_me: '0',
      });
    }
  }

  render() {
    return this.props.isLoading ? 'Loading....' : this.props.children;
  }
}

export default Auth;
