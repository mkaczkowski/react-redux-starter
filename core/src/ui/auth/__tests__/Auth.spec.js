import React from 'react';
import { render, cleanup } from 'react-testing-library';
import Auth from '../Auth';

describe('Auth - dumb', () => {
  afterEach(cleanup);

  it('should display loader', () => {
    const { container } = render(
      <Auth isLoading={true}>
        <div />
      </Auth>
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should display content', () => {
    const { container } = render(
      <Auth isLoading={false}>
        <div />
      </Auth>
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
