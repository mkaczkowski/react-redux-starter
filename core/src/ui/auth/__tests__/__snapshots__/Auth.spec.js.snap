// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Auth - dumb should display content 1`] = `<div />`;

exports[`Auth - dumb should display loader 1`] = `Loading....`;
